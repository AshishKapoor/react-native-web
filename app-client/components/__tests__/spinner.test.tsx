import React from "react";
import renderer, { ReactTestRendererJSON, create } from "react-test-renderer";
import LoadingSpinner from "../spinner";

import { TamaguiProvider } from "tamagui";
import config from "../../tamagui.config";

jest.mock("expo-router", () => ({ Link: "Link" }));

describe("Components/Spinner", () => {
  const SpinnerComponent = (
    <TamaguiProvider config={config}>
      <LoadingSpinner />
    </TamaguiProvider>
  );

  it("renders correctly", () => {
    const spinner = create(SpinnerComponent).toJSON() as ReactTestRendererJSON;
    const activityIndicator = spinner?.children![0];
    expect(spinner?.type!).toBe("View");
    expect(activityIndicator?.children![0].type!).toBe("ActivityIndicator");
    expect(activityIndicator?.props!.testID).toBe("activity-indicator");
    expect(activityIndicator?.children![0].props!.size).toBe("large");
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<LoadingSpinner />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
