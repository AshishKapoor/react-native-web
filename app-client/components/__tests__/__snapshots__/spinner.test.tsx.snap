// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Spinner matches snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "paddingBottom": 13,
      "paddingLeft": 13,
      "paddingRight": 13,
      "paddingTop": 13,
    }
  }
>
  <View
    style={
      {
        "alignItems": "stretch",
        "flexDirection": "column",
      }
    }
    testID="activity-indicator"
  >
    <ActivityIndicator
      color=""
      size="large"
    />
  </View>
</View>
`;
